subprojects/gradle/cd/src/main/kotlin/com/avito/ci/steps/FlakyReportStep.kt

```diff
@@ -0,0 +1,38 @@
+package com.avito.ci.steps
+
+import com.android.build.gradle.internal.tasks.factory.dependsOn
+import com.avito.impact.configuration.internalModule
+import com.avito.instrumentation.extractReportCoordinates
+import com.avito.instrumentation.instrumentationTask
+import com.avito.test.summary.flakyReportTask
+import com.avito.test.summary.testSummaryPluginId
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.tasks.TaskProvider
+
+class FlakyReportStep(context: String, name: String) : SuppressibleBuildStep(context, name),
+    ImpactAnalysisAwareBuildStep by ImpactAnalysisAwareBuildStep.Impl() {
+
+    var configuration: String = ""
+
+    override fun registerTask(project: Project, rootTask: TaskProvider<out Task>) {
+        require(project.pluginManager.hasPlugin(testSummaryPluginId)) {
+            "FlakyReport step can't be initialized without $testSummaryPluginId plugin applied"
+        }
+
+        require(configuration.isNotBlank()) {
+            "FlakyReport step can't be initialized without provided instrumentation configuration"
+        }
+
+        if (useImpactAnalysis && !project.internalModule.isModified()) return
```

Comment:
Also, I'd prefer to skip the task but have it in the graph.

Reply:
Also, `onlyIf` is executed in an execution phase. We shouldn't avoid using heavy impact analysis in a configuration.

Reply:
well, i don't like this way to handle impact analysis as well; but seeing at as out of this pr scope change
It's how we deal with impact analysis in all steps

Reply:
Agree